File: A4/src/command/TextMenu.java
package command;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

import model.MyStmtException;

public class TextMenu
{
	private Map<String, Command> commands;
	public TextMenu()
	{
		commands = new HashMap<>();
	}
	public void addCommand(Command c)
	{
		commands.put(c.getKey(), c);
	}
	public void printMenu()
	{
		for(Command com: commands.values())
		{
			String line = String.format("%4s: %s",  com.getKey(), com.getDescription());
			System.out.println(line);
		}
	}
	public void show() throws IOException, MyStmtException
	{
		Scanner scanner = new Scanner(System.in);
		while(true)
		{
			printMenu();
			System.out.printf("Input the option: ");
			String key = scanner.nextLine();
			Command com = commands.get(key);
			if (com == null)
			{
				System.out.println("Invalid Option");
				continue;
			}
			//System.out.println(key);
			//System.out.println(com.getDescription());
			com.execute();
		}
	}
}

File: A7/Model/TextMenu.cs
using A7.Dictionary;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Model
{
    class TextMenu
    {
        private MyIDictionary<String, Command> commands;
        public TextMenu()
        {
            commands = new MyDictionary<String, Command>();
        }
        public void addCommand(Command c)
        {
            commands.Add(c.GetKey(), c);
        }
        public void PrintMenu()
        {
            foreach (Command com in commands.Values())
            {
                String line = "    " + com.GetKey() + ": " + com.GetDescription();
                Console.WriteLine(line);
            }
        }
        public void show()
        {
            while (true)
            {
                PrintMenu();
                Console.Write("Input the option: ");
                String key = Console.ReadLine();
                Command com = commands.Lookup(key);
                if (com == null)
                {
                    Console.WriteLine("Invalid Option");
                    continue;
                }
                com.Execute();
            }
        }
    }
}
